src/main/java/io/vertx/httpproxy/interceptors/BodyInterceptor.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2011-2024 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.httpproxy.interceptors;
+
+import io.vertx.codegen.annotations.GenIgnore;
+import static io.vertx.codegen.annotations.GenIgnore.PERMITTED_TYPE;
+
+import io.vertx.codegen.annotations.Unstable;
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonObject;
+import io.vertx.httpproxy.ProxyInterceptor;
+import io.vertx.httpproxy.interceptors.impl.BodyInterceptorImpl;
+
+import java.util.function.Function;
+
+/**
+ * Used to create interceptors to modify request and response bodies.
+ */
+@VertxGen
+@Unstable
+public interface BodyInterceptor {
```

Comment:
Can you take a look at the Vert.x Web Client's `BodyCodec` API? I think you could inspire from it to reduce the number of methods in this class to a minimum. The user should be able to choose between `Buffer`, JSON or POJO.